src/resolver/virustotal.rs

```diff
@@ -0,0 +1,109 @@
+    resolutions: Option<Vec<Resolution>>,
+    detected_url_count: usize,
+    detected_downloaded_sample_count: usize,
+    undetected_downloaded_sample_count: usize,
+    undetected_url_count: usize,
+}
+
+impl VTResolver {
+    pub fn new() -> Self {
+        Self
+    }
+
+    fn request(ip: IpAddr) -> anyhow::Result<String> {
+        let api_key = std::env::var("VIRUSTOTAL_API_KEY")
+            .map_err(|_| anyhow::Error::msg("VIRUSTOTAL_API_KEY not set in environment"))?;
+        let url = format!("https://www.virustotal.com/vtapi/v2/ip-address/report?apikey={}&ip={}", api_key, ip);
+        let client = Client::new();
+        let response = client.get(&url).send()?;
+        Ok(response.text()?)
+    }
+
+    fn parse(response: &str) -> anyhow::Result<VirusTotalRecord> {
+        let raw_record: RawVirusTotalRecord = serde_json::from_str(response)?;
+        let vt_record = VirusTotalRecord {
+            response_code: raw_record.response_code,
+            verbose_msg: raw_record.verbose_msg,
+            asn: raw_record.asn,
+            country: raw_record.country,
+            resolutions: raw_record.resolutions,
+            detected_url_count: raw_record.detected_urls.map_or(0, |v| v.len()),
+            detected_downloaded_sample_count: raw_record.detected_downloaded_samples.map_or(0, |v| v.len()),
+            undetected_downloaded_sample_count: raw_record.undetected_downloaded_samples.map_or(0, |v| v.len()),
+            undetected_url_count: raw_record.undetected_urls.map_or(0, |v| v.len()),
+        };
+        Ok(vt_record)
+    }
+
+    pub fn resolve(&self, ip_addresses: Vec<IpAddr>) -> anyhow::Result<Vec<String>> {
+        let results = Vec::new();
+        for ip in ip_addresses {
+            let response = Self::request(ip)?;
+            let vt_record = Self::parse(&response)?;
+
+            println!("VirusTotal Record for: {:?}", ip);
+            println!("Response code: {:?}", vt_record.response_code);
+            println!("Verbose message: {:?}", vt_record.verbose_msg);
+            println!("ASN: {:?}", vt_record.asn);
+            println!("Country: {:?}", vt_record.country);
+            println!("Detected URL count: {:?}", vt_record.detected_url_count);
+            println!("Detected downloaded sample count: {:?}", vt_record.detected_downloaded_sample_count);
+            println!("Undetected downloaded sample count: {:?}", vt_record.undetected_downloaded_sample_count);
+            println!("Undetected URL count: {:?}", vt_record.undetected_url_count);
+
+            if let Some(resolutions) = &vt_record.resolutions {
+                if !resolutions.is_empty() {
+                    for resolution in resolutions.iter().take(5) {
+                        println!("{:?}", resolution);
+                    }
+                } else {
+                    println!("No resolutions found");
+                }
+            }
+
+            println!();
+        }
+        Ok(results)
+    }
+}
```

Comment:
Let's add a tests module to this, at least for the parse method and we can figure out the best way to add tests for the VT API since it requires an API key.